src/Select.tsx

```diff
@@ -16,15 +17,35 @@ export * from './generated/Select.js';
 
 export type SelectReactRendererProps = ReactSimpleRendererProps<SelectElement>;
 
+type SelectRenderer = ComponentType<SelectReactRendererProps>;
+
 export type SelectProps = Partial<Omit<_SelectProps, 'children' | 'renderer'>> &
   Readonly<{
-    children?: ReactNode | ComponentType<SelectReactRendererProps>;
-    renderer?: ComponentType<SelectReactRendererProps> | null;
+    children?: ReactNode | SelectRenderer | Array<ReactNode | SelectRenderer>;
+    renderer?: SelectRenderer | null;
   }>;
 
 function Select(props: SelectProps, ref: ForwardedRef<SelectElement>): ReactElement | null {
+  const children = Children.toArray(props.children as ReactNode[]);
+
+  // Components with slot attribute should stay in light DOM.
+  const slotted = children.filter((child) => {
+    return typeof child === 'object' && 'props' in child && child.props.slot;
```

Comment:
You can use [React.isValidElement](https://react.dev/reference/react/isValidElement) here.

Reply:
Thanks, done.